File: Data/Models/Comment.cs
using System.ComponentModel.DataAnnotations;

namespace AngularCore.Data.Models
{
    public class Comment : BaseEntity
    {
        [Required]
        public User User { get; set; }

        [Required]
        public Post Post { get; set; }

        [Required]
        [MaxLength(256)]
        public string Content { get; set; }
    }
}

File: ClientApp/src/app/models/comment.ts
import { User } from '@app/models/user';
import { Post } from '@app/models/post';

export class Comment {

  constructor(
    public id: string,
    public author: User,
    public post: Post,
    public content: string,
    public createdAt: Date,
    public modifiedAt: Date
  ){}
}
